roborazzi-accessibility-check/src/main/java/com/github/takahirom/roborazzi/ATFAccessibilityChecker.kt

```diff
@@ -0,0 +1,179 @@
+import com.google.android.apps.common.testing.accessibility.framework.AccessibilityViewCheckResult
+import com.google.android.apps.common.testing.accessibility.framework.Parameters
+import com.google.android.apps.common.testing.accessibility.framework.ViewChecker
+import com.google.android.apps.common.testing.accessibility.framework.integrations.espresso.AccessibilityViewCheckException
+import com.google.android.apps.common.testing.accessibility.framework.utils.contrast.BitmapImage
+import com.google.common.collect.ImmutableSet
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers
+import org.robolectric.shadows.ShadowBuild
+
+@ExperimentalRoborazziApi
+data class ATFAccessibilityChecker(
+  val checks: Set<AccessibilityHierarchyCheck> = AccessibilityCheckPreset.getAccessibilityHierarchyChecksForPreset(
+    AccessibilityCheckPreset.LATEST
+  ),
+  val suppressions: Matcher<in AccessibilityViewCheckResult> = Matchers.not(Matchers.anything()),
+) {
+  constructor(
+    preset: AccessibilityCheckPreset,
+    suppressions: Matcher<in AccessibilityViewCheckResult> = Matchers.not(Matchers.anything()),
+  ) : this(
+    checks = AccessibilityCheckPreset.getAccessibilityHierarchyChecksForPreset(preset),
+    suppressions = suppressions,
+  )
+
+
+  @RequiresApi(34)
+  internal fun runAllChecks(
+    view: View,
+    screenshotBitmap: Bitmap?,
+    checks: Set<AccessibilityHierarchyCheck>,
+    suppressions: Matcher<in AccessibilityViewCheckResult>,
+  ): List<AccessibilityViewCheckResult> {
+    val parameters = Parameters().apply {
+      if (screenshotBitmap != null) {
+        putScreenCapture(BitmapImage(screenshotBitmap))
+      }
+      setSaveViewImages(true)
+    }
+
+    val viewChecker = ViewChecker().apply {
+      setObtainCharacterLocations(true)
+    }
+
+    val results = viewChecker.runChecksOnView(ImmutableSet.copyOf(checks), view, parameters)
+
+    return results.filter {
+      !suppressions.matches(it)
+    }
+  }
+
+  internal fun runAccessibilityChecks(
+    captureRoot: CaptureRoot,
+    roborazziOptions: RoborazziOptions,
+    failureLevel: CheckLevel,
+  ) {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+      roborazziErrorLog("Skipping accessibilityChecks on API " + Build.VERSION.SDK_INT + "(< ${Build.VERSION_CODES.UPSIDE_DOWN_CAKE})")
+      return
+    }
+
+    if (Build.FINGERPRINT == "robolectric") {
+      // TODO remove this once ATF doesn't bail out
+      // https://github.com/google/Accessibility-Test-Framework-for-Android/blob/c65cab02b2a845c29c3da100d6adefd345a144e3/src/main/java/com/google/android/apps/common/testing/accessibility/framework/uielement/AccessibilityHierarchyAndroid.java#L667
+      ShadowBuild.setFingerprint("roborazzi")
+    }
+
+    if (captureRoot is CaptureRoot.Compose) {
+      val view = (captureRoot.semanticsNodeInteraction.fetchSemanticsNode().root as ViewRootForTest).view.rootView
```

Comment:
Can we remove `.rootView` to enable this at the component level, as specified by the user in the Rule?

Reply:
I think the .root before that is a problem also.

Maybe a follow up. I suspect it will need filtering instead.

But I can try. I'd prefer to tackle incrementally if it doesn't just work by removing that.

Reply:
OK, your suggestion seemed to work. But I can only demonstrate it by running at a smaller level, the current codepath runs it at the end of the test on the screen.

See `composableOnly` in the test.